File: lib/model/video_pick_result.dart
part of kang_image_picker;

@immutable
class VideoPickResult {
  final String videoPath;
  final String videoFilename;
  final String thumbnailPath;
  final String thumbnailFilename;
  final int thumbnailWidth;
  final int thumbnailHeight;
  final double duration;
  final int size;

  // const VideoPickResult({
  //   required this.videoPath,
  //   required this.thumbnailPath,
  //   required this.thumbnailWidth,
  //   required this.thumbnailHeight,
  //   required this.duration,
  // });

  VideoPickResult.fromJson(Map json)
      : videoPath = json['videoPath'] as String,
        videoFilename = json['videoFilename'] as String,
        thumbnailPath = json['thumbnailPath'] as String,
        thumbnailFilename = json['thumbnailFilename'] as String,
        thumbnailWidth = json['thumbnailWidth'] as int,
        thumbnailHeight = json['thumbnailHeight'] as int,
        duration = json['duration'] as double,
        size = json['size'] as int;

  Map<String, dynamic> toJson() => <String, dynamic>{
        'videoPath': videoPath,
        'videoFilename': videoFilename,
        'thumbnailPath': thumbnailPath,
        'thumbnailFilename': thumbnailFilename,
        'thumbnailWidth': thumbnailWidth,
        'thumbnailHeight': thumbnailHeight,
        'duration': duration,
        'size': size,
      };

  @override
  String toString() {
    return 'VideoPickResult{videoPath: $videoPath,videoFilename:$videoFilename, thumbnailPath: $thumbnailPath,thumbnailFilename: $thumbnailFilename, thumbnailWidth: $thumbnailWidth, thumbnailHeight: $thumbnailHeight, duration: $duration}';
  }
}

File: android/src/main/java/com/jalagar/kang_image_picker/VideoPickResult.java
package com.jalagar.kang_image_picker;

import java.util.HashMap;
import java.util.Map;

public class VideoPickResult {
    private String videoPath;

    private String videoFilename;
    private Double duration;

    private String thumbnailFilename;
    private String thumbnailPath;
    private int thumbnailWidth;
    private int thumbnailHeight;

    private Long size;

    public String getVideoPath() {
        return videoPath;
    }

    public void setVideoPath(String videoPath) {
        this.videoPath = videoPath;
    }

    public String getVideoFilename() {
        return videoFilename;
    }

    public void setVideoFilename(String videoFilename) {
        this.videoFilename = videoFilename;
    }

    public Double getDuration() {
        return duration;
    }

    public void setDuration(Double duration) {
        this.duration = duration;
    }

    public String getThumbnailFilename() {
        return thumbnailFilename;
    }

    public void setThumbnailFilename(String thumbnailFilename) {
        this.thumbnailFilename = thumbnailFilename;
    }

    public String getThumbnailPath() {
        return thumbnailPath;
    }

    public void setThumbnailPath(String thumbnailPath) {
        this.thumbnailPath = thumbnailPath;
    }

    public int getThumbnailWidth() {
        return thumbnailWidth;
    }

    public void setThumbnailWidth(int thumbnailWidth) {
        this.thumbnailWidth = thumbnailWidth;
    }

    public int getThumbnailHeight() {
        return thumbnailHeight;
    }

    public void setThumbnailHeight(int thumbnailHeight) {
        this.thumbnailHeight = thumbnailHeight;
    }

    public Long getSize() {
        return size;
    }

    public void setSize(Long size) {
        this.size = size;
    }

    public Map<String, Object> toMap() {
        return new HashMap<String, Object>() {
            {
                put("videoPath", videoPath);
                put("videoFilename", videoFilename);
                put("thumbnailFilename", thumbnailFilename);
                put("thumbnailPath", thumbnailPath);
                put("thumbnailWidth", thumbnailWidth);
                put("thumbnailHeight", thumbnailHeight);
                put("duration", duration);
                put("size", size);
            }
        };
    }
}
